File: java/bogosort.java
public static class Program {
    static void bogoSort(int[] array, int printIterationsEvery) {
        int iteration = 0;
        printIteration(array, iteration);
        while (!isSorted(array)) {
            iteration++;
            array = shuffle(array);
            if (iteration % printIterationsEvery == 0) 
                printIteration(array, iteration);
        }
        System.out.println("Result:");
        printIteration(array, iteration);
    }
    static int[] shuffle(int[] array) {
        for (int i = 0; i < array.length; i++) {
            int j = (int)(Math.random() * (i + 1));
            int temp = array[i];
            array[i] = array[j];
            array[j] = temp;
        }
        return array;
    }
    static boolean isSorted(int[] array) {
        for (int i = 1; i < array.length; i++) {
            if (array[i - 1] > array[i]) return false;
        }
        return true;
    }
    static int[] generateRandomArray(int length) {
        int[] array = new int[length];
        for (int i = 0; i < length; i++) {
            array[i] = -100 + (int)(Math.random() * 201)
        }
        return array;
    }
    static void printIteration(int[] array, int iteration) {
        String arrayText = "";
        for (int i = 0; i < array.length; i++){
            arrayText += array[i];
            if (i != array.length - 1) arrayText += ", ";
        }
        System.out.println(String.format("%d, [%s]", iteration, arrayText));
    }
    public static void main(String[] args) {
        bogoSort(generateRandomArray(8), 100);
    }
}

File: javascript/bogosort.js

BogoSort(GenerateRandomArray(8), 100);

function BogoSort(array, printIterationsEvery) {
    var iteration = 0;
    PrintIteration(array);
    while (!IsSorted(array)) {
        iteration++;
        array = Shuffle(array);
        if (iteration % printIterationsEvery == 0) { 
            PrintIteration(array, iteration);
        }
    }
    console.log("Result:");
    PrintIteration(array, iteration);
}
function Shuffle(array) {
    var c = array.length, i, temp;
    while (c > 0) {
        i = Math.floor(Math.random() * c);
        c--;
        temp = array[c];
        array[c] = array[i];
        array[i] = temp;
    }
    return array;
}
function IsSorted(array) {
    for (i = 1; i < array.length; i++)
    {
        if (array[i - 1] > array[i]) { return false; }
    }
    return true;
}
function GenerateRandomArray(length) {
    return Array.from({length: length}, () => -100 + Math.floor(Math.random() * 201));
}
function PrintIteration(array, iteration = 0) {
    console.log(`${iteration}, [${array}]`);
}